test/abstractarray.jl

```diff
@@ -709,3 +709,16 @@ let
     @test !issparse(m1)
     @test !issparse(m2)
 end
+
+#isinteger and isreal
+@test isinteger(Diagonal(rand(1:5,5)))
+@test isreal(Diagonal(rand(5)))
```

Comment:
ah. seems like one of those vectorized things we could maybe think about deprecating since `all(isinteger, x)` should work fine